packages/react/src/use-check-resource.ts

```diff
@@ -0,0 +1,70 @@
+import type {
+  CheckResourceRequest,
+  CheckResourcesResult,
+  RequestOptions,
+} from "@cerbos/core";
+// eslint-disable-next-line import/no-extraneous-dependencies -- TODO: not sure about this rule, maybe importing from peer deps should be allowed?
+import isEqual from "lodash/isEqual";
+// eslint-disable-next-line import/no-extraneous-dependencies -- TODO: not sure about this rule, maybe importing from peer deps should be allowed?
+import { useCallback, useEffect, useRef, useState } from "react";
+
+import { useCerbos } from "./use-cerbos";
+
+export interface UseCheckResourceReturn {
+  isLoading: boolean;
+  data: CheckResourcesResult | undefined;
+  error: Error | undefined;
+}
+
+export type UseCheckResourceParams = Omit<CheckResourceRequest, "principal">;
```

Comment:
Let's just inline these `Omit` types, otherwise we end up adding a whole extra page to the docs for each of them.